camel-core/src/main/java/org/apache/camel/processor/aggregate/AggregateProcessor.java

```diff
@@ -1448,6 +1448,11 @@ protected void doStop() throws Exception {
         if (recoverService != null) {
             camelContext.getExecutorServiceManager().shutdown(recoverService);
         }
+
+        if (shutdownTimeoutCheckerExecutorService) {
+            camelContext.getExecutorServiceManager().shutdown(timeoutCheckerExecutorService);
```

Comment:
Looks like tests require to set executor service to null and flag to false after shutdown

Reply:
Does not seem that simple. And I need to be doubly careful to touch the tests

Reply:
@gsudharsan , yes, you can try. I did the changes and pass mvn clean install with success...